projects/packages/masterbar/src/admin-menu/class-base-admin-menu.php

```diff
@@ -0,0 +1,797 @@
+			add_submenu_page( $slug, $menu_item[3], $menu_item[0], $menu_item[1], $url, null, 0 );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Updates the submenus of the given menu slug.
+	 *
+	 * It hides the menu by adding the `hide-if-js` css class and duplicates the submenu with the new slug.
+	 *
+	 * @param string $slug Menu slug.
+	 * @param array  $submenus_to_update Array of new submenu slugs.
+	 */
+	public function update_submenus( $slug, $submenus_to_update ) {
+		global $submenu;
+
+		if ( ! isset( $submenu[ $slug ] ) ) {
+			return;
+		}
+
+		// This is needed for cases when the submenus to update have the same new slug.
+		$submenus_to_update = array_filter(
+			$submenus_to_update,
+			static function ( $item, $old_slug ) {
+				return $item !== $old_slug;
+			},
+			ARRAY_FILTER_USE_BOTH
+		);
+
+		/**
+		 * Iterate over all submenu items and add the hide the submenus with CSS classes.
+		 * This is done separately of the second foreach because the position of the submenu might change.
+		 */
+		foreach ( $submenu[ $slug ] as $index => $item ) {
+			if ( ! array_key_exists( $item[2], $submenus_to_update ) ) {
+				continue;
+			}
+
+			$this->hide_submenu_element( $index, $slug, $item );
+		}
+
+		$submenu_items = array_values( $submenu[ $slug ] );
+
+		/**
+		 * Iterate again over the submenu array. We need a copy of the array because add_submenu_page will add new elements
+		 * to submenu array that might cause an infinite loop.
+		 */
+		foreach ( $submenu_items as $i => $submenu_item ) {
+			if ( ! array_key_exists( $submenu_item[2], $submenus_to_update ) ) {
+				continue;
+			}
+
+			add_submenu_page(
+				$slug,
+				isset( $submenu_item[3] ) ? $submenu_item[3] : '',
+				isset( $submenu_item[0] ) ? $submenu_item[0] : '',
+				isset( $submenu_item[1] ) ? $submenu_item[1] : 'read',
+				$submenus_to_update[ $submenu_item[2] ],
+				'',
+				0 === $i ? 0 : $i + 1
+			);
+		}
+	}
+
+	/**
+	 * Adds a menu separator.
+	 *
+	 * @param int    $position The position in the menu order this item should appear.
+	 * @param string $cap Optional. The capability required for this menu to be displayed to the user.
+	 *                         Default: 'read'.
+	 */
+	public function add_admin_menu_separator( $position = null, $cap = 'read' ) {
+		$menu_item = array(
+			'',                                  // Menu title (ignored).
+			$cap,                                // Required capability.
+			wp_unique_id( 'separator-custom-' ), // URL or file (ignored, but must be unique).
+			'',                                  // Page title (ignored).
+			'wp-menu-separator',                 // CSS class. Identifies this item as a separator.
+		);
+
+		$this->set_menu_item( $menu_item, $position );
+	}
+
+	/**
+	 * Enqueues scripts and styles.
+	 */
+	public function enqueue_scripts() {
+		$assets_base_path = '../../build/admin-menu/';
+		if ( $this->is_rtl() ) {
+			$css_path = 'admin-menu.rtl.css';
```

Comment:
I tried to be consistent with the module code here but I'm not sure this is right. My understanding is that we should choose one of the following approaches:
- A custom webpack config so `rtl` files are named like `-rtl.css` instead of `.rtl.css`: In this case we shouldn't have to load the actual file but only use `wp_style_add_data( 'jetpack-admin-menu', 'rtl', 'replace' );`
- Assuming `rtl` files keep the `.rtl.css` name then check if `is_rtl()` as we do currently and load the file manually without using `wp_style_add_data`

Please let me know if I got something wrong here.

Reply:
I saw @anomiex 's [comment on another PR](https://github.com/Automattic/jetpack/pull/37339#discussion_r1598584806) and I think the best approach is to:
- Keep the default webpack config when it comes to creating RTL styles, aka  `.rtl.css`
- Use `Assets::register_script` to load js and css dependencies.

That said, @anomiex I was wondering if you could take a look at the webpack config in this PR and share some insights. Since lots of @Automattic/jetpack-vulcan team members are currently working on relevant tasks maybe we could standardize how we handle minifying assets and using webpack (the `rtl` approach is an example of that). Then we could P2 about it too so that everyone can benefit from the lessons learned here.

Reply:
I gave the webpack config a once-over and left a comment in there.